server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -3581,9 +3562,29 @@ public Vlan createVlanAndPublicIpRange(final long zoneId, final long networkId,
                     if (!NetUtils.is31PrefixCidr(newCidr)) {
                         if (NetUtils.ipRangesOverlap(startIP, endIP, otherVlanStartIP, otherVlanEndIP)) {
                             throw new InvalidParameterValueException("The IP range already has IPs that overlap with the new range." +
-                                " Please specify a different start IP/end IP.");
+                                    " Please specify a different start IP/end IP.");
                         }
                     }
+                } else {
+                    // For tagged or non-overlapping URIs we need to ensure there is no Public traffic type
+                    boolean overlapped = false;
+                    if( network.getTrafficType() == TrafficType.Public ) {
```

Comment:
Code formatting for this section of code is different ( spaces used )

Reply:
Do you mean ` if( network.getTrafficType()....` should be reformatted to ` if (network.getTrafficType()....`?